File: sample/src/main/java/com/nearit/sample/CustomFirebaseInstanceIdService.java
package com.nearit.sample;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

import it.near.sdk.push.fcmregistration.NearInstanceIDListenerService;


public class CustomFirebaseInstanceIdService extends FirebaseInstanceIdService {

    @Override
    public void onTokenRefresh() {
        String refreshedToken = FirebaseInstanceId.getInstance().getToken();
        NearInstanceIDListenerService.sendRegistrationToServer(refreshedToken);
    }
}

File: sample-kotlin/src/main/java/com/nearit/sample_kotlin/CustomFirebaseInstanceIdService.kt
package com.nearit.sample_kotlin

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import it.near.sdk.push.fcmregistration.NearInstanceIDListenerService

class CustomFirebaseInstanceIdService : FirebaseInstanceIdService() {

    override fun onTokenRefresh() {
        val refreshedToken: String? = FirebaseInstanceId.getInstance().token
        NearInstanceIDListenerService.sendRegistrationToServer(refreshedToken)
    }

}